SampleGallery/Samples/SDK 17134/RadialGradients/RadialGradients.xaml.cs

```diff
@@ -0,0 +1,310 @@
+                _changeButtonGradientStop1.Duration = TimeSpan.FromSeconds(2);
+
+                _BBGradientStop1.StartAnimation(nameof(_BBGradientStop1.Color), _changeButtonGradientStop1);
+
+                _changeButtonGradientStop2 = _compositor.CreateColorKeyFrameAnimation();
+                _changeButtonGradientStop2.InsertKeyFrame(0, _orangeRed);
+                _changeButtonGradientStop2.InsertKeyFrame(1, _lightBlue);
+                _changeButtonGradientStop2.Duration = TimeSpan.FromSeconds(2);
+
+                _BBGradientStop2.StartAnimation(nameof(_BBGradientStop2.Color), _changeButtonGradientStop2);
+
+                // Animations for the pulsing effect are turned off
+                _stop1OffsetAnim = _compositor.CreateScalarKeyFrameAnimation();
+                _stop1OffsetAnim.InsertKeyFrame(0, 0);
+                _stop1OffsetAnim.InsertKeyFrame(0f, 0f);
+                _stop1OffsetAnim.Duration = TimeSpan.FromSeconds(1.5f);
+                _stop1OffsetAnim.IterationCount = 50;
+                
+                _PBGradientStop1.StartAnimation("Offset", _stop1OffsetAnim);
+
+                _stop2offsetAnim2 = _compositor.CreateScalarKeyFrameAnimation();
+                _stop2offsetAnim2.InsertKeyFrame(0, 0);
+                _stop2offsetAnim2.InsertKeyFrame(0f, 0f);
+                _stop2offsetAnim2.Duration = TimeSpan.FromSeconds(1.5f);
+                _stop2offsetAnim2.IterationCount = 50;
+                _stop2offsetAnim2.DelayTime = TimeSpan.FromSeconds(.25f);
+
+                _PBGradientStop2.StartAnimation("Offset", _stop2offsetAnim2);
+
+                _pulseColor = _compositor.CreateColorKeyFrameAnimation();
+                _pulseColor.InsertKeyFrame(0, _transparent);
+                _pulseColor.InsertKeyFrame(1, _transparent);
+                _pulseColor.Duration = TimeSpan.FromSeconds(1.5);
+
+                _PBGradientStop1.StartAnimation(nameof(_PBGradientStop1.Color), _pulseColor);
+
+                // Create scale animation the helps with the final visual end result of create a pulsing effect
+                _scale = _compositor.CreateVector3KeyFrameAnimation();
+                _scale.InsertKeyFrame(0, new Vector3(0, 0, 0));
+                _scale.InsertKeyFrame(1, new Vector3(1, 1, 0));
+                _scale.Duration = TimeSpan.FromSeconds(1.5);
+                _scale.IterationCount = 50;
+
+                _pulseVisual.StartAnimation("Scale", _scale);
+
+                // Turn off the stopwatch and change text back to Start
+                _dt.Stop();
+                _stopwatch.Start();
+                Timer.Content = ("Start");
+
+                // Set our boolean flag for the stopwatch animation to false to indicate it's off
+                animationOn = false;
+            }
+        }
+
+        // Method that sets up the stopwatch to display the time or "Start" when clicked on and off
+        private void Dt_Tick(object sender, object e)
```

Comment:
Is your intention to display the time continuosly as it updates? Check how much CPU this is using on your machine. It seems like a bad idea because you'll end up using one core at 100% just to update a display that can't possibly update more than once per frame. There is no point updating this more than 20 times per seconds, and even that is probably overkill.

Reply:
A lot of the code for the dispatcher timer and stopwatch were pulling directly from various documentation or online examples, so I'm not sure what exactly you're talking about here - should I remove the method?

Reply:
Depends. What are you trying to have it do?

Reply:
From what I'm remembering as I wrote this, I thought this was the "tick" method that made the stopwatch count

Reply:
To follow up - I did not have CPU ramp up and the use of DT and stopwatch together seems the correct way to write this in UWP.

Reply:
updated